sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -494,7 +497,7 @@ public String getId() {
      * @param partitionKey the partition key
      * @return an {@link Mono} containing the  cosmos item resource response.
      */
-    public Mono<CosmosAsyncItemResponse> deleteItem(String itemId, PartitionKey partitionKey) {
+    public Mono<CosmosAsyncItemResponse<Object>> deleteItem(String itemId, PartitionKey partitionKey) {
```

Comment:
API change for @kushagraThapar and @moderakh to review.

Reply:
Yes, this looks good to me. 
@mbhaskar  - I noticed we have some APIs, that do not take classType in parameter, like this one. Is it okay to have type as `Object` for these ?

Reply:
The item in case of deleteItem would be null. so Object should work